### DISCUSSION TITLE
Automerge not working on self-hosted Renovate if cache is not set up

### DISCUSSION THREAD

**Discussion by zicdamasta** [renovatebot/renovate#26532] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

v39.2.4

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

None

### Describe the problem

We are using self-hosted Renovate v39.2.4 running in [GitHub Action](https://github.com/renovatebot/github-action).
It is not documented anywhere, but if caching is not set up, you should set [ignorePrAuthor ](https://docs.renovatebot.com/configuration-options/#ignoreprauthor)as `true`.
Otherwise it tries to find open PRs [from the cache](https://github.com/renovatebot/renovate/blob/a17713c7c3c50cff72b5423a1362b81f4c757b7c/lib/modules/platform/github/index.ts#L788C29-L788C29), but there are no cache and it always returns 0 results.

### Relevant debug logs

<details><summary>Logs</summary>

```

DEBUG: getBranchPr(renovate/actions-checkout-4.x) (repository=acme/reponame)
DEBUG: findPr(renovate/actions-checkout-4.x, undefined, open) (repository=acme/reponame)
DEBUG: findPr(renovate/actions-checkout-4.x, undefined, closed) (repository=acme/reponame)
DEBUG: 0 PRs are currently open (repository=acme/reponame)

```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by astellingwerf** [comment]
The version number you mention makes no sense. The planning for version 38 started in August of last year, an 39 in November, but neither of these have been released. Can you double check what version you're using?

You state that you "have linked to a minimal reproduction in the description above", but I don't see it listed. Could you elaborate where to find it?

**Reply by rarkins** [reply]
Likely the GitHub action version, which diverged

**Comment by rarkins** [comment]
Your claims aren't likely to be true as automerge does work without cache for many people. 

You might have the "username" configured incorrectly - especially likely if you're using an app token. 

Try that or provide all initial action logs

**Reply by HonkingGoose** [reply]
> It is not documented anywhere, but if caching is not set up, you should set [ignorePrAuthor ](https://docs.renovatebot.com/configuration-options/#ignoreprauthor) as true.
>
> > Your claims aren't likely to be true as automerge does work without cache for many people.

I skimmed [our Self-hosting docs](https://docs.renovatebot.com/self-hosted-configuration/) for the `cach` search term. It looks like Renovate creates its own default cache directory, if the user does nothing.

Also the [`ignorePrAuthor` docs](https://docs.renovatebot.com/configuration-options/#ignoreprauthor) say that you should only set `ignorePrAuthor` to `true` for a short while to help when you're migrating from the Mend Renovate app to self-hosted. After that migration is done you should remove the `ignorePrAuthor` line from the config to revert back to the default behavior, which is `ignorePrAuthor=false`.

So as far as I can see from our docs, and @rarkins response, it looks like the docs are correct?

> ## ignorePrAuthor
> 
> This is usually needed if someone needs to migrate bot accounts, including from the Mend Renovate App to self-hosted.
> If `ignorePrAuthor` is configured to true, it means Renovate will fetch the entire list of repository PRs instead of optimizing to fetch only those PRs which it created itself.
> You should only want to enable this if you are changing the bot account (e.g. from `@old-bot` to `@new-bot`) and want `@new-bot` to find and update any existing PRs created by `@old-bot`.
> It's recommended to revert this setting once that transition period is over and all old PRs are resolved.

**Comment by zicdamasta** [comment]
> Your claims aren't likely to be true as automerge does work without cache for many people.

It works with automergeType=branch, because you don't need to run action second time to find already opened PR.

> You might have the "username" configured incorrectly - especially likely if you're using an app token.

username is set correctly using this guide https://github.com/orgs/community/discussions/24664#discussioncomment-3880274.
We are using GitHub App and App token.

> Try that or provide all initial action logs

I attached logs. [logs.zip](https://github.com/renovatebot/renovate/files/13866651/logs.zip)
run1 is state, when there are 0 open PR's - creates new PR. All okay.
run2 is a state, when there are opened PR from run1, but it does not find it. In the logs you can see that Renovate even tries to recreate PR, but fails.

**Reply by rarkins** [reply, marked as answer]
I still suspect that your configured `username` is wrong. Change `acme-Renovate[bot]` to all lower case (`acme-renovate[bot]`).

**Reply by zicdamasta** [reply]
> I still suspect that your configured `username` is wrong. Change `acme-Renovate[bot]` to all lower case (`acme-renovate[bot]`).

It worked! Thank you.
